core/control-plane/control-plane-api/src/main/java/org/eclipse/dataspaceconnector/controlplane/api/ControlPlaneApiExtension.java

```diff
@@ -0,0 +1,125 @@
+/*
+ *  Copyright (c) 2020, 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Microsoft Corporation - initial API and implementation
+ *
+ */
+
+package org.eclipse.dataspaceconnector.controlplane.api;
+
+import org.eclipse.dataspaceconnector.api.auth.AuthenticationRequestFilter;
+import org.eclipse.dataspaceconnector.api.auth.AuthenticationService;
+import org.eclipse.dataspaceconnector.controlplane.api.transferprocess.TransferProcessControlApiController;
+import org.eclipse.dataspaceconnector.controlplane.api.transferprocess.model.TransferProcessFailStateDto;
+import org.eclipse.dataspaceconnector.runtime.metamodel.annotation.Extension;
+import org.eclipse.dataspaceconnector.runtime.metamodel.annotation.Inject;
+import org.eclipse.dataspaceconnector.runtime.metamodel.annotation.Provider;
+import org.eclipse.dataspaceconnector.runtime.metamodel.annotation.Provides;
+import org.eclipse.dataspaceconnector.spi.EdcException;
+import org.eclipse.dataspaceconnector.spi.WebServer;
+import org.eclipse.dataspaceconnector.spi.WebService;
+import org.eclipse.dataspaceconnector.spi.system.Hostname;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtension;
+import org.eclipse.dataspaceconnector.spi.system.ServiceExtensionContext;
+import org.eclipse.dataspaceconnector.spi.transfer.TransferProcessManager;
+import org.eclipse.dataspaceconnector.spi.transfer.callback.ControlPlaneApiUrl;
+import org.jetbrains.annotations.NotNull;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import static java.lang.String.format;
+
+/**
+ * {@link ControlPlaneApiExtension } exposes HTTP endpoints for internal interaction with the Control Plane
+ */
+@Provides(ControlPlaneApiUrl.class)
+@Extension(value = ControlPlaneApiExtension.NAME)
+public class ControlPlaneApiExtension implements ServiceExtension {
+
+    public static final String NAME = "Control Plane API";
+
+    public static final String CONTROL_PLANE_API_CONFIG = "web.http.controlplane";
+    public static final String CONTROL_PLANE_CONTEXT_ALIAS = "controlplane";
```

Comment:
Maybe "control" would be more generic and fit with the context already defined by the data-plane (DataPlaneExtension), so if control-plane and data-plane are deployed together the control api will be exposed on the same context.
I would change it also at line 49

Reply:
I am afraid this would break some existing behaviors... As here it would lead to registering the injected `authService` under the `control` context (see l. 94), while this context is already used when the data plane validates the access tokens against the control plane. See here:

https://user-images.githubusercontent.com/88310985/157418442-da5cc5aa-6b54-42b3-b836-e4f0a3f46fcf.png

Reply:
@bscholtes1A yes, you're right, this could fall in the same context of [the comment I left about the `AuthenticationService`](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/pull/2116#discussion_r1003121732), maybe the authentication should be related to the context so we could have an authentication dedicated to the intra-connector communication. To be honest for the moment I would remove the authentication on this extension and create a dedicated issue for it.

Reply:
Completely agreed, let's remove the Auth Service for now and uniformize this Control Plane API later on.

Reply:
Ok so do you propose to remove the auth filter and also from the client remove the auth key right?

Reply:
Yes, since this API is not supposed to be exposed to the outside, as it is only for intra-connector communication, I don't see it as a big issue to leave it without auth for now, we should bring endpoint authentication to an higher level

Reply:
@wolf4ood  could you please rename the context to `control` now that the AuthService has been removed?